File: src/main/python/sorting/bogosort/bogo_sort.py
from random import randrange


def bogo_sort(to_sort):
    while not is_sorted(to_sort):
        shuffle(to_sort)


def shuffle(to_sort):
    for i in range(len(to_sort)):
        rand_index = randrange(0, len(to_sort))
        to_sort[i], to_sort[rand_index] = to_sort[rand_index], to_sort[i]


def is_sorted(to_sort):
    for i in range(1, len(to_sort)):
        if to_sort[i] < to_sort[i - 1]:
            return False

    return True

File: src/main/java/sorting/bogosort/BogoSort.java
package sorting.bogosort;

public class BogoSort {

    public static void sort(int[] arr) {
        while (!isSorted(arr)) {
            shuffle(arr);
        }
    }

    private static void shuffle(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            swap(arr, i, (int) (Math.random() * arr.length));
        }
    }

    private static void swap(int[] arr, int i, int j) {
        int temp = arr[i];
        arr[i] = arr[j];
        arr[j] = temp;
    }

    private static boolean isSorted(int[] arr) {
        int i = arr.length;

        while (i-- > 1) {
            if (arr[i] < arr[i - 1]) {
                return false;
            }
        }

        return true;
    }
}
